app/app.go

```diff
@@ -158,20 +169,28 @@ func (app *MerkleEyesApp) doTx(tree merkle.Tree, tx []byte) abci.Result {
 
 func (app *MerkleEyesApp) Commit() abci.Result {
 
-	hash := app.state.Commit()
-
+	app.hash = app.state.Hash()
 	app.height++
+	fmt.Printf("height=%d,hash=%x\n", app.height, app.hash)
+
 	if app.db != nil {
-		app.db.Set(eyesStateKey, wire.BinaryBytes(MerkleEyesState{
-			Hash:   hash,
+		// This is in the batch with the Save, but not in the tree
+		tree := app.state.Append().(*iavl.IAVLTree)
```

Comment:
But the tree is immutable. So can't we save the tree first, and then save this state stuff to the db? If we crash on restart, all we lose on the restart is the reference to the new tree, but we won't lose any data, from what I can tell. I think it would make for simpler code, but not sure if it's actually better - but that's what I did for the persistent dummy